deploy/olm-catalog/smart-gateway-operator/manifests/smartgateway.infra.watch_smartgateways_crd.yaml

```diff
@@ -34,75 +34,130 @@ spec:
-                Defaults to using 'messaging-internal-<name>.<namespace>.src:5672/collectd/telemetry'.
-              type: string
+            applications:
+              description: Applications to load.
+              items:
+                properties:
+                  config:
+                    description: Free form configuration for the application.
+                    type: string
+                  name:
+                    description: Name of the application.
+                    type: string
+              type: array
+            bridge:
+              description: AMQP1 bridge configuration options
+              properties:
+                amqpUrl:
+                  description: URL of the AMQP endpoint. Format of ^amqp://(([a-z]+)(:([a-z]+))*@)*([a-zA-Z_0-9.-]+)(:([0-9]+))*(.+)$.
+                    Defaults to amqp://127.0.0.1:5672/collectd/telemetry.
+                  pattern: ^amqp://(([a-z]+)(:([a-z]+))*@)*([a-zA-Z_0-9.-]+)(:([0-9]+))*(.+)$
+                  type: string
+                containerIdPattern:
+                  description: AMQP container ID (should be unique). Defaults to "sa-%x".
+                  type: string
+                enabled:
+                  description: Whether to enable the bridge for connection to AMQP.
+                    Defaults to true.
+                  type: boolean
+                inetTarget:
+                  description: Connect to gateway with inet socket. Defaults to 127.0.0.1:30000.
+                  type: string
+                ringBufferCount:
+                  description: Number of message buffers between AMQP and Outgoing.
+                    Defaults to 15000.
+                  type: integer
+                ringBufferSize:
+                  description: Size of a message buffer between AMQP and Outgoing.
+                    Defaults to 2048.
+                  type: integer
+                socketBlock:
+                  description: Outgoing socket connection will block. Defaults to
+                    true.
+                  type: boolean
+                statsPeriod:
+                  description: How often to print stats, 0 for no stats. Value is
+                    in seconds. Defaults to 60.
+                  type: integer
+                stopCount:
+                  description: Number of AMQP mesg to rcv before exit, 0 for continous.
+                    Defaults to 0.
+                  type: integer
+                unixSocketPath:
+                  description: Connect to gateway with unix socket. Defaults to /tmp
+                    with 'smartgateway' appended as the final path, e.g. /tmp/smartgateway.
+                  type: string
+                verbose:
+                  description: Print extra info, multiple instance increase verbosity.
+                    Defaults to true.
+                  type: boolean
+              type: object
```

Comment:
Instead of trying to be clever by parsing the `transport` configuration, make the `bridge` container a top level managed object instead of inferring it from the transport. In this way you can (theoretically) start up just a bridge, or not use it at all if your transport is not a `socket`. Allows better control in the deployment.